### ISSUE TITLE
viz: relax `l5d-viz-pods-running` condition when using HA mode

### ISSUE THREAD

**Issue by mateiidavid** [linkerd/linkerd2#7385]
## Background

As part of #7232 I investigated whether linkerd would still behave correctly in the event of a node outage, when installed in `--ha` mode. Everything worked mostly as expected, however, I bumped into a minor issue with `viz` that I thought would be interesting to talk about.

For the purpose of the investigation, I created a `k3d` cluster with three nodes and installed everything as normal. I then removed the node to check what would happen. During this whole investigation, no new nodes were added. 

A third of all linkerd pods were in the end descheduled, since the node went down. It did take a few minutes, however, for k8s to reconcile state. My pods were errenously marked as `Running` for a long while, even after the node they were hosted on went into a `NotReady` state. I was issuing queries to the `metrics-api` through `viz stat` and everything seemed to work well. After k8s caught up on its new state changes and descheduled the pods, `viz stat` no longer worked:

```
:; linkerd viz stat deploy -n emojivoto
Cannot connect to Linkerd Viz: The "tap-f86d97c48-hcsb6" pod is not running
Validate the install with: linkerd viz check
```

The reason why it no longer worked didn't have to do with the `metrics-api` being down (there were still 2 pods available), but with the check we are running when trying to connect to the api:

https://github.com/linkerd/linkerd2/blob/b91f385a1a796d4434242f7abb8aa833786be2b0/viz/pkg/api/api.go#L25-L40

Among other things, we check that the viz extension pods are all running:

https://github.com/linkerd/linkerd2/blob/b91f385a1a796d4434242f7abb8aa833786be2b0/viz/pkg/healthcheck/healthcheck.go#L150-L168

This seems like the correct thing to do under normal circumstances, but perhaps in certain scenarios (i.e `ha`) we could have a minimum number of pods required to run?

### Solution

I'm first and foremost interested to see what others think, and if it would be a good idea to change the check. We generally recommend people run at least four nodes in their cluster when running in `--ha` mode. If I would have added a replacement node in the cluster, the state would have been eventually reconciled, and all viz deployment pods would have been running.

However, the `viz` stack is meant to help users troubleshoot their traffic and cluster. I think in cases like this, it would be valuable to have access to viz CLI commands. This would also prevent us from opening up a dashboard, even though the dashboard _still works_. In my case, I had the dashboard open for the entire duration of the investigation, if I would have closed it and tried to re-open it after k8s updated its state and descheduled the nodes, I wouldn't have been able to.

**Solution**: I haven't spent a whole lot of time researching what we could do to improve this. `viz` has its own `Healthchecker` which is a wrapper around the core `Healthchecker`. We could perhaps wire up the ha boolean value into it and use that when checking whether all pods are running. As an alternative, we could probably also get the deployment's replica count and pass the check if at least 25% of the deployment is running, or something of that sort.

Interested to hear your thoughts on this and if it'd be a good thing to do :) 

**TL;DR**: when running in HA mode, if 1/3 pods are available we should still be able to use `viz` cli commands.
